src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs

```diff
@@ -243,6 +293,61 @@ public string ToString(string directory)
             return string.Join(System.Environment.NewLine, lines.ToArray());
         }
 
+        /// <summary>
+        /// Returns the string representation of the match object same format as ToString()
+        /// and inverts the color of the matched text if virtual terminal is supported.
+        /// </summary>
+        /// <param name="directory">Directory to use as the root when calculating the relative path.</param>
+        /// <returns>The string representation of the match object with matched text inverted.</returns>
+        public string ToEmphasizedString(string directory)
+        {
+            if (!_emphasize)
+            {
+                return ToString(directory);
+            }
+
+            return ToString(directory, EmphasizeLine());
+        }
+
+        /// <summary>
+        /// Surrounds the matched text with virtual terminal sequences to invert it's color. Used in ToEmphasizedString.
+        /// </summary>
+        /// <returns>The matched line with matched text inverted.</returns>
+        private string EmphasizeLine()
```

Comment:
Good candidate to be static.

Reply:
But it uses `_matchIndexes`?

Reply:
We can pass `_matchIndexes` by parameter.

Reply:
I agree. Just like a whole bunch of other PowerShell code that is similarly caught up by stateful object state implicitly passed in to methods.

But I believe that pre-existed here and this contribution shouldn't be held back on the basis of code debt incurred before PowerShell was open-sourced. This PR has been open long enough and has been a good, responsive, constrained contribution and I think we really need to just accept or reject it, rather than turning it into a Ship of Theseus.

Reply:
Yeah, I think this one's gone on long enough and is a definite nice-to-have in PSv7 😁

Reply:
The PR is delayed because it is in new area for PowerShell. However, the code should be clean.

Reply:
Since this isn't a public API, we can always refactor internal code.  I would suggest a separate PR if it's a concern or @iSazonov you can always add a commit to this PR if you feel strongly about this one.

Reply:
Yes, I can :-) Do you ok that we have over 70 opened PR? :-)

Reply:
@iSazonov I'll ask the team to start driving down the number of open PRs